std/bitmanip.d

```diff
@@ -1014,9 +1128,146 @@ public:
         }
     }
 
+    /**
+     * Returns: true if all the bits in this `BitArray` are set to true.
+     */
+    @property bool all() const @nogc pure nothrow
+    {
+        if (_len == 0)
+            return false;
+        foreach (i; 0 .. dim - 1)
+            if (~_ptr[i] != 0)
+                return false;
+        return (_ptr[dim - 1] & endMask) == endMask;
+    }
+
+    @system unittest
+    {
+        debug(bitarray) printf("BitArray.all.unittest\n");
+        BitArray empty;
+        assert(!empty.all);
+        auto a = BitArray([1]);
+        auto b = BitArray([1, 1, 1, 1, 1]);
+        auto c = BitArray([1, 1, 1, 1, 0, 1, 1]);
+        auto d = BitArray([1, 1, 1, 0]);
+        auto multiWord = BitArray([1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+            1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+            1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+            1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1]);
+        assert(a.all);
+        assert(b.all);
+        assert(!c.all);
+        assert(!d.all);
+        assert(!multiWord.all);
+    }
+
+    /**
+     * Returns: true if any of the bits in this `BitArray` is set to true.
+     */
+    @property bool any() const @nogc pure nothrow
+    {
+        if (_len == 0)
+            return false;
+        foreach (i; 0 .. dim - 1)
+            if (_ptr[i] != 0)
+                return true;
+        return (_ptr[dim - 1] & endMask) != 0;
+    }
+
+    @system unittest
+    {
+        debug(bitarray) printf("BitArray.any.unittest\n");
+        BitArray empty;
+        assert(!empty.any);
+        auto a = BitArray([0]);
+        auto b = BitArray([0, 0, 0, 1, 0, 1]);
+        auto c = BitArray([1, 0, 0, 0, 0, 0]);
+        auto d = BitArray([0, 0, 0, 0, 0, 0, 0, 0]);
+        BitArray e;
+        e.length = 260;
+        e[259] = true;
+        assert(!a.any);
+        assert(b.any);
+        assert(c.any);
+        assert(!d.any);
+        assert(e.any);
+    }
+
+    /**
+     * Returns: true if none of the bits in this `BitArray` is set to true.
+     */
+    @property bool none() const @nogc pure nothrow
+    {
+        return !any;
+    }
+
+    @system unittest
+    {
+        debug(bitarray) printf("BitArray.none.unittest\n");
+        BitArray empty;
+        assert(empty.none);
+        auto a = BitArray([0]);
+        auto b = BitArray([0, 0, 0, 1, 0, 1]);
+        auto c = BitArray([1, 0, 0, 0, 0, 0]);
+        auto d = BitArray([0, 0, 0, 0, 0, 0, 0, 0]);
+        BitArray e;
+        e.length = 260;
+        e[259] = true;
+        assert(a.none);
+        assert(!b.none);
+        assert(!c.none);
+        assert(d.none);
+        assert(!e.none);
+    }
+
+    /**
+     * Returns: Number of bits set to true.
+     */
+    @property size_t count() const @nogc pure nothrow
```

Comment:
On the contrary - it's possible to make much faster implementation of these primitives using BitArray internals.